crates/bevy_render/src/mesh/mesh.rs

```diff
@@ -115,54 +127,48 @@ pub enum Indices {
 #[derive(Debug)]
 pub struct Mesh {
     pub primitive_topology: PrimitiveTopology,
-    pub attributes: Vec<VertexAttribute>,
+    pub attributes: Vec<VertexAttributeData>,
     pub indices: Option<Indices>,
+    pub attribute_vertex_buffer_descriptor: Option<VertexBufferDescriptor>, //TODO julian: should not be Option
 }
 
 impl Mesh {
-    pub fn new(primitive_topology: PrimitiveTopology) -> Self {
+    pub fn new(
+        primitive_topology: PrimitiveTopology,
+        mut attributes: Vec<VertexAttributeData>,
+        indices: Option<Indices>,
+    ) -> Self {
+        let mut mesh = Mesh {
+            primitive_topology,
+            attributes: Vec::default(),
+            indices,
+            attribute_vertex_buffer_descriptor: None,
+        };
+        for attribute in attributes.drain(..) {
+            mesh.insert_attribute(attribute);
+        }
+        mesh
```

Comment:
```suggestion
        Mesh {
            primitive_topology,
            attributes,
            indices,
            attribute_vertex_buffer_descriptor: None,
        }
```

Reply:
Since `Mesh::insert_attribute` just pushes the input into the `attributes` field on a `Mesh`, it's equivalent to just making the initial `attributes` field the same vec.

Reply:
Assuming this is correct, I would suggest renaming the below `new_empty` back to `new` (as it functionally was before) to encourage users to construct the `Mesh` in-place, which is the more bevy way to do it (in general).